packages/vitest/src/node/core.ts

```diff
@@ -724,6 +725,18 @@ export class Vitest {
       this.report('onProcessTimeout').then(() => {
         console.warn(`close timed out after ${this.config.teardownTimeout}ms`)
         this.state.getProcessTimeoutCauses().forEach(cause => console.warn(cause))
+
+        if (!this.pool) {
+          const runningServers = [this.server, ...this.projects.map(p => p.server)].filter(Boolean).length
+
+          if (runningServers === 1)
+            console.warn('Tests closed successfully but something prevents Vite server from exiting')
```

Comment:
Is there a way to know exactly what's preventing Vite from exiting? @AriPerkkio

Reply:
In #2008 there are mentions of ESLint Vite plugin, `@vitejs/plugin-react-swc`, custom Vite plugins using chokidar, ESbuild, etc. Most of these can attach listeners for file system changes. When these are combined with coverage, the amount of work these plugins do when they pick up the coverage report directory can be heavy.

I don't think there is any easy way to detect what exactly is causing these errors. The `hanging-process` reporter should give the best estimate to users.

Reply:
Okay.

Reply:
If vitest isn't able to know exactly what is preventing it from exiting, it maybe worth while having a message with a more explicit call to action of what to look for to help guide users to self fix the issue. Example:

```suggestion
            console.warn('Tests closed successfully but something prevents Vite server from exiting. Some common culprits include plugins which might attach file system change listeners which aren't properly cleaned up, [MORE EXAMPLES HERE].')
```

Maybe even some instructions about how to use the `hanging-process` reporter to help debug where it maybe occurring.

Reply:
I think it's a nice idea to give some examples and/or a link to the issue to help debug it.